v-next/hardhat/src/internal/cli/init/init.ts

```diff
@@ -1,36 +1,184 @@
+    }
+
+    throw e;
   }
+}
 
-  // Many terminal emulators in windows fail to run the createProject()
-  // workflow, and don't present themselves as TTYs. If we are in this
-  // situation we throw a special error instructing the user to use WSL or
-  // powershell to initialize the project.
-  if (process.platform === "win32") {
-    throw new HardhatError(
-      HardhatError.ERRORS.GENERAL.NOT_INSIDE_PROJECT_ON_WINDOWS,
+// generated with the "colossal" font
+function printAsciiLogo() {
+  const logoLines = `
+888    888                      888 888               888
+888    888                      888 888               888
+888    888                      888 888               888
+8888888888  8888b.  888d888 .d88888 88888b.   8888b.  888888
+888    888     "88b 888P"  d88" 888 888 "88b     "88b 888
+888    888 .d888888 888    888  888 888  888 .d888888 888
+888    888 888  888 888    Y88b 888 888  888 888  888 Y88b.
+888    888 "Y888888 888     "Y88888 888  888 "Y888888  "Y888
+`.trim();
+
+  console.log(chalk.blue(logoLines));
+}
+
+// NOTE: This function is exported for testing purposes
+export async function printWelcomeMessage(): Promise<void> {
+  const hardhatVersion = await getHardhatVersion();
+
+  console.log(
+    chalk.cyan(`👷 Welcome to ${HARDHAT_NAME} v${hardhatVersion} 👷\n`),
+  );
+
+  // Warn the user if they are using an outdated version of Hardhat
+  try {
+    const latestHardhatVersion = await getLatestHardhatVersion();
+    if (hardhatVersion !== latestHardhatVersion) {
+      console.warn(
+        chalk.yellow.bold(
+          `⚠️ You are using an outdated version of Hardhat. The latest version is v${latestHardhatVersion}. Please consider upgrading to the latest version before continuing with the project initialization. ⚠️\n`,
+        ),
+      );
+    }
+  } catch (e) {
+    console.warn(
+      chalk.yellow.bold(
+        `⚠️ We couldn't check if you are using the latest version of Hardhat. Please consider upgrading to the latest version if you are not using it yet. ⚠️\n`,
+      ),
     );
   }
-
-  throw new HardhatError(HardhatError.ERRORS.GENERAL.NOT_IN_INTERACTIVE_SHELL);
 }
 
-async function throwIfCwdAlreadyInsideProject() {
+/**
+ * getWorkspace asks the user for the workspace to initialize the project in
+ * if the input workspace is undefined.
+ *
+ * It also validates that the workspace is not already initialized.
+ *
+ * NOTE: This function is exported for testing purposes
+ *
+ * @param workspace The path to the workspace to initialize the project in.
+ * @returns The path to the workspace.
+ */
+export async function getWorkspace(workspace?: string): Promise<string> {
+  // Ask the user for the workspace to initialize the project in if it was not provided
+  if (workspace === undefined) {
+    workspace = await promptForWorkspace();
+  }
+
+  workspace = path.resolve(workspace);
+
+  if (!(await exists(workspace))) {
+    throw new HardhatError(HardhatError.ERRORS.GENERAL.UNSUPPORTED_OPERATION, {
```

Comment:
This should have its own error descriptor

Reply:
Of course! It also helped me find another place where I used `UNSUPPORTED_OPERATION` where in fact we already had a more specific error available. Fixed both now.